### ISSUE TITLE
Build failing for quick start [Tried Vert.x health check and Http] - OSIO intermittently cannot reach OSO

### ISSUE THREAD

**Issue by arunkumars08** [openshiftio/openshift.io#3299]
I was trying to create a build using quick starts (Legacy flow)

Steps:

1. Reset my environment
2. Updated my tenant
3. Created a new space
4. Using the legacy wizard flow, tried creating a quick start
5. Picked Vert.x HEalth check
6. All successful and the pipeline started
7. During the time of release, build failed with the following error.

<img width="1666" alt="screen shot 2018-04-25 at 2 33 49 pm" src="https://user-images.githubusercontent.com/7880465/39236278-fefdf0a4-4895-11e8-8a5d-f99d5f04264c.png">

I am not sure if I am the only one facing this problem.

**Comment by piyush1594** [comment]
This failed at the deploy stage and seems that it is not able to reach OpenShift.

**Comment by chmouel** [comment]
Updated BZ https://bugzilla.redhat.com/show_bug.cgi?id=1552373

**Comment by qodfathr** [comment]
I am currently blocked by this. As this is now resulting in service degradation (at least for me and a few others), I'm tempted to move to SEV1. Thoughts?

**Comment by ldimaggi** [comment]
+1 to Todd's comment - is this limited to selected users? or is it based on the overall system load, or do we know?

**Comment by piyush1594** [comment]
@ldimaggi Is this fixed or closed because of a duplicate of some other issue?

**Comment by ldimaggi** [comment]
Neither - thanks for noticing! - this was closed in error.

**Comment by bmicklea** [comment]
Looks like we have two +1s for moving to SEV1, please lower if justified (and explain).

**Comment by jaseemabid** [comment]
I've seen this once in a while and since its a network/infrastructure issue, there is nothing much build team could do. In general the services underneath are flaky and when you see a lot of these, you start to ignore them. 

I think we should monitor the infra bits more carefully and fix it at the lower level, it is substantially harder to fix these things when the real problem is hidden deep inside the stack trace of a random Jenkins plugin none of us is familiar with.

**Comment by pradeepto** [comment]
cc @krishnapaparaju  @maxandersen

**Comment by piyush1594** [comment]
@maxandersen @krishnapaparaju @pradeepto Can you please take a look at this?

**Comment by pradeepto** [comment]
possibly related to #3058 

(from the upstream bugz link)

>
Can any who have this issue do whats suggesting in bugzilla:
Can you curl the pod IP addresses directly?
Can you curl the service while checking tcpdump output?
Can you attach the OVS flows and iptables rules please?

oc adm diagnostics networkcheck

@bmicklea @qodfathr This issue is so intermittent and we can't reproduce it. We tried it many times. The upstream folks have asked for those

**Comment by krishnapaparaju** [comment]
@qodfathr this is an intermittent problem, I guess you going thru (no building getting started) is tracked here at #3422

**Comment by bmicklea** [comment]
Do we have a sense for how quickly it goes away?  Or once it happens does it somehow "corrupt" a user's project/space?

**Comment by pradeepto** [comment]
@krishnapaparaju No, @qodfathr 's is facing a different issue. Todd is on an old tenant Jenkins which still doesn't have the problem that users complained on #3422 .  

As mentioned, this issue is different and is very intermittent and shows all signs of network/infra issue. If seen again, please provide the above data. We need to pass that on to upstream folks. 

@rohanKanojia @qodfathr @ldimaggi @arunkumars08  Please take a note.

**Comment by jfchevrette** [comment]
If someone sees a "No route to host" error again, I will need the following information sent out to me ASAP (in addition to commenting here, I'd suggest you ping me on chat.openshift.io)

`oc -n namespace get pods -ocustom-columns='NS:.metadata.namespace,NAME:.metadata.name,NODE:.spec.nodeName'`

If you dont know how/where to run that command, please reach out to me directly on chat.openshift.io and I can get the info I need and have upstream investigate at the network layers ASAP

**Comment by pradeepto** [comment]
@lordofthejars @rohanKanojia Can you please update the issue with the commands mentioned in the comment https://github.com/openshiftio/openshift.io/issues/3299#issuecomment-386275622 and JF's comment https://github.com/openshiftio/openshift.io/issues/3299#issuecomment-386600399 .

**Comment by mmclanerh** [comment]
are folks still experiencing this? my impression is yes --  if so, let's try to get some debug data captured as described via https://github.com/openshiftio/openshift.io/issues/3299#issuecomment-386600399

**Comment by piyush1594** [comment]
Issue #3058 seems like also oso networking and I commented there with details asked. Here's the link https://github.com/openshiftio/openshift.io/issues/3058#issuecomment-386805721

**Comment by piyush1594** [comment]
and if you want to reproduce this, then PR https://github.com/fabric8-services/fabric8-tenant-jenkins/pull/88 will always result into #3058

**Comment by lordofthejars** [comment]
I have not been able to reproduce in my case.

**Comment by piyush1594** [comment]
Hey @lordofthejars, try PR https://github.com/fabric8-services/fabric8-tenant-jenkins/pull/88 and you will always result into #3058 and that is also regarding openshift networking or something related to openshift

**Comment by lordofthejars** [comment]
@piyush1594 So we cannot do anything right? So we could un-assign this from our team?

**Comment by piyush1594** [comment]
@lordofthejars I think nothing we(build-team) can do at this stage for this issue and #3058 but if you go through the discussion on #3058 you will get a better idea.

**Comment by bmicklea** [comment]
So in this case if the problem is in openshift itself how do we ensure it's being worked to completion on that side? Is there a case open in Bugzilla?

**Comment by jfchevrette** [comment]
The more concrete details / reproducers we get the better upstream can investigate and pinpoint the root cause of this. So please keep sending details when you hit this issue.

**Comment by jfchevrette** [comment]
That being said, I haven't had any reports in about a week.

**Comment by jaseemabid** [comment]
@jfchevrette The build team is using a service and we just happened to be the people discovering these issues. Its not fair to expect us to debug and fix them as well. This is most probably an openshift networking problem, which happens during build once in a while because we make a ton of network calls. I'm not sure we can actually do anything more on this. 

This is a potential SEV1 and P0, but it really shouldn't be build-cd. 

cc: @krishnapaparaju

**Comment by mmclanerh** [comment]
further debug details will help not only with upstream artifacts, but also around design/architecture challenges we might encounter through concurrent operations.

if a dependency is unavailable (eg: rolling a new deployment, affinitize a pod to a new node),  we should think about how the caller behaves (and ultimately, what that looks like to the user).  how should the app deal with a dependency unavailable (retry? bankruptcy?).. do we properly log these events to the errortracking service?   it looks like we may be retrying here.. but it's hard to tell, how quickly that's happening.  if this is waiting a total of 2m30s and it takes 3m for a volume release/recycle to happen, that could be a potential race condition, right?

perhaps we should also think on the deployment strategy as configured for the related pods -- in cases where we are using exclusive write (rwo) volumes and stateful data, do we need to change parameters of the deployment strategy or perhaps the grace period for termination..?

**Comment by jfchevrette** [comment]
@jaseemabid ok to reassign this to the service delivery team. However having received no reports of this still occurring in the past 1+ week I'm tempted to consider this resolved. Looking at the original report the error state that jenkins could not reach the content-repository pod through the internal service route and that was fixed about a week ago by making sure content-repository is unidled before jenkins is started. (#3469)

**Comment by jaseemabid** [comment]
@mmclanerh I don't disagree. From what we find here, the whole logic of idler might be broken. https://github.com/fabric8-services/fabric8-jenkins-idler/issues/233#issuecomment-387370518

**Comment by jfchevrette** [comment]
Thanks for reporting this @piyush1594 !

I have added the details to upstream BZ https://bugzilla.redhat.com/show_bug.cgi?id=1552373

**Comment by sthaha** [comment]
Removed `team/build-cd` since it seems like an upstream openshift issue and nothing related to pipeline.

**Comment by mvulavak** [comment]
@jfchevrette Is this still an issue?

**Comment by jfchevrette** [comment]
@mvulavak not to my knowledge

**Comment by chmouel** [comment]
I think it's not an issue because we removed the content-repository and that's not affecting us as much anymore, let there be glory until we have a new container introduced in the jenkins namespace. we'll reopen if needed,
